### ISSUE TITLE
Jruby fails on one of the ruby decryption tests

### ISSUE THREAD

**Issue by llimllib** [department-of-veterans-affairs/connect_vbms#11]
https://travis-ci.org/department-of-veterans-affairs/connect_vbms/jobs/76146034#L269

```
1) Ruby Encrypt/Decrypt test vs Java reference impl encrypts in java, and decrypts using ruby
   Failure/Error: decrypted_xml = VBMS.decrypt_message_xml_ruby(encrypted_xml, test_pc12_server_key,
   RuntimeError:
     org.w3c.dom.DOMException: NOT_SUPPORTED_ERR: The implementation does not support the requested type of object or operation. 
   # nokogiri/XmlNode.java:1662:in `add_child_node'
   # ./vendor/bundle/jruby/2.2.0/gems/nokogiri-1.6.6.2-java/lib/nokogiri/xml/node.rb:814:in `add_child_node_and_reparent_attrs'
   # ./vendor/bundle/jruby/2.2.0/gems/nokogiri-1.6.6.2-java/lib/nokogiri/xml/node.rb:125:in `add_child'
   # ./vendor/bundle/jruby/2.2.0/gems/nokogiri-1.6.6.2-java/lib/nokogiri/xml/node.rb:405:in `parent='
   # ./vendor/bundle/jruby/2.2.0/gems/nokogiri-1.6.6.2-java/lib/nokogiri/xml/document_fragment.rb:25:in `block in initialize'
   # ./vendor/bundle/jruby/2.2.0/gems/nokogiri-1.6.6.2-java/lib/nokogiri/xml/node_set.rb:187:in `block in each'
   # ./vendor/bundle/jruby/2.2.0/gems/nokogiri-1.6.6.2-java/lib/nokogiri/xml/node_set.rb:186:in `each'
   # ./vendor/bundle/jruby/2.2.0/gems/nokogiri-1.6.6.2-java/lib/nokogiri/xml/document_fragment.rb:25:in `initialize'
   # nokogiri/XmlDocumentFragment.java:107:in `new'
   # ./vendor/bundle/jruby/2.2.0/gems/nokogiri-1.6.6.2-java/lib/nokogiri/xml/node.rb:355:in `fragment'
   # ./vendor/bundle/jruby/2.2.0/gems/nokogiri-1.6.6.2-java/lib/nokogiri/xml/node.rb:796:in `coerce'
   # ./vendor/bundle/jruby/2.2.0/gems/nokogiri-1.6.6.2-java/lib/nokogiri/xml/node.rb:257:in `replace'
   # ./vendor/bundle/jruby/2.2.0/gems/xmlenc-0.2.0/lib/xmlenc/encrypted_data.rb:39:in `decrypt'
   # ./vendor/bundle/jruby/2.2.0/gems/xmlenc-0.2.0/lib/xmlenc/encrypted_document.rb:21:in `block in decrypt'
   # ./vendor/bundle/jruby/2.2.0/gems/xmlenc-0.2.0/lib/xmlenc/encrypted_document.rb:18:in `decrypt'
   # ./src/vbms/common.rb:78:in `decrypt_message_xml_ruby'
   # ./spec/ruby_crypto_spec.rb:35:in `block in (root)'
```

**Comment by harrisj** [comment]
Getting a different jruby issue now

```
/usr/local/var/rbenv/versions/jruby-1.7.20/bin/jruby -I/Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/rspec-core-3.3.2/lib:/Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/rspec-support-3.3.0/lib /Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/rspec-core-3.3.2/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb
Coverage may be inaccurate; Try setting JRUBY_OPTS="-Xcli.debug=true --debug"
/Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/simplecov-0.10.0/lib/simplecov.rb:43 warning: tracing (e.g. set_trace_func) will not capture all events without --debug flag
SyntaxError: /Users/jacobharris/code/18f/connect_vbms/src/vbms/client.rb:6: syntax error, unexpected tLABEL
    def self.from_env_vars(logger: nil, env_name: 'test')
                                  ^
          require at org/jruby/RubyKernel.java:1072
           (root) at /Users/jacobharris/code/18f/connect_vbms/src/vbms.rb:1
          require at org/jruby/RubyKernel.java:1072
           (root) at /Users/jacobharris/code/18f/connect_vbms/src/vbms.rb:12
          require at org/jruby/RubyKernel.java:1072
           (root) at /Users/jacobharris/code/18f/connect_vbms/spec/spec_helper.rb:1
             load at org/jruby/RubyKernel.java:1091
           (root) at /Users/jacobharris/code/18f/connect_vbms/spec/spec_helper.rb:10
             each at org/jruby/RubyArray.java:1613
           (root) at /Users/jacobharris/code/18f/connect_vbms/spec/client_spec.rb:1
           (root) at /Users/jacobharris/code/18f/connect_vbms/spec/client_spec.rb:1
           (root) at /Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/rspec-core-3.3.2/lib/rspec/core/configuration.rb:1
  load_spec_files at /Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/rspec-core-3.3.2/lib/rspec/core/configuration.rb:1327
  load_spec_files at /Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/rspec-core-3.3.2/lib/rspec/core/configuration.rb:1325
           (root) at /Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/rspec-core-3.3.2/exe/rspec:4
Coverage report generated for RSpec to /Users/jacobharris/code/18f/connect_vbms/coverage. 37 / 76 LOC (48.68%) covered.
```

**Comment by alex** [comment]
Oh, you'll need to run with `JRUBY_OPTS="--2.0"`

On Mon, Sep 14, 2015 at 1:52 PM, Jacob Harris <notifications@github.com>
wrote:

> Getting a different jruby issue now
>
> /usr/local/var/rbenv/versions/jruby-1.7.20/bin/jruby -I/Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/rspec-core-3.3.2/lib:/Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/rspec-support-3.3.0/lib /Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/rspec-core-3.3.2/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb
> Coverage may be inaccurate; Try setting JRUBY_OPTS="-Xcli.debug=true --debug"
> /Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/simplecov-0.10.0/lib/simplecov.rb:43 warning: tracing (e.g. set_trace_func) will not capture all events without --debug flag
> SyntaxError: /Users/jacobharris/code/18f/connect_vbms/src/vbms/client.rb:6: syntax error, unexpected tLABEL
>     def self.from_env_vars(logger: nil, env_name: 'test')
>                                   ^
>           require at org/jruby/RubyKernel.java:1072
>            (root) at /Users/jacobharris/code/18f/connect_vbms/src/vbms.rb:1
>           require at org/jruby/RubyKernel.java:1072
>            (root) at /Users/jacobharris/code/18f/connect_vbms/src/vbms.rb:12
>           require at org/jruby/RubyKernel.java:1072
>            (root) at /Users/jacobharris/code/18f/connect_vbms/spec/spec_helper.rb:1
>              load at org/jruby/RubyKernel.java:1091
>            (root) at /Users/jacobharris/code/18f/connect_vbms/spec/spec_helper.rb:10
>              each at org/jruby/RubyArray.java:1613
>            (root) at /Users/jacobharris/code/18f/connect_vbms/spec/client_spec.rb:1
>            (root) at /Users/jacobharris/code/18f/connect_vbms/spec/client_spec.rb:1
>            (root) at /Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/rspec-core-3.3.2/lib/rspec/core/configuration.rb:1
>   load_spec_files at /Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/rspec-core-3.3.2/lib/rspec/core/configuration.rb:1327
>   load_spec_files at /Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/rspec-core-3.3.2/lib/rspec/core/configuration.rb:1325
>            (root) at /Users/jacobharris/code/18f/connect_vbms/vendor/bundle/jruby/1.9/gems/rspec-core-3.3.2/exe/rspec:4
> Coverage report generated for RSpec to /Users/jacobharris/code/18f/connect_vbms/coverage. 37 / 76 LOC (48.68%) covered.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/department-of-veterans-affairs/connect_vbms/issues/11#issuecomment-140158814>
> .
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084

**Comment by harrisj** [comment]
Thank you. I should've asked before I posted this

**Comment by harrisj** [comment]
Basically, it throws an exception somewhere within this method

```
    /**
     * Add <code>other</code> as a child of <code>this</code>.
     */
    @JRubyMethod(visibility=Visibility.PRIVATE)
    public IRubyObject add_child_node(ThreadContext context, IRubyObject other) {
        return adoptAs(context, AdoptScheme.CHILD, other);
    }
```

but the stack trace does not provide deeper illumination. Looking for a JRuby-compatible debugger

**Comment by harrisj** [comment]
Fixed in PR #74

**Comment by amoose** [comment]
Completed with #74
